pydicom/filewriter.py

```diff
@@ -284,7 +291,7 @@ def write_DA(fp, data_element, padding=' '):
     if isinstance(val, (str, compat.string_types)):
         write_string(fp, data_element, padding)
     else:
-        if isinstance(val, (list, tuple)):
+        if is_multi_value(val):
```

Comment:
this appears several times, maybe you could refactory it with a private function `_awesome_name_task` that takes `_format_TM`, `_format_DT` etc... and then add a test.

maybe something like 
```py
def _my_awesome_format(format, val):
    if is_multi_value(val):
￼              return = "\\".join((x if isinstance(x, (str, compat.string_types))
￼                               else _format_DT(x) for x in val))
    else:
          return format(val)
```
```py
@pytest.mark.parametrize("format_function,expected", [
    (_format_TM, 'correctanser'),
    (_format_DA, 'correctanser'),
...
])
def test_my_awesome_formating_not_multivalue(format_func, expected):
    val = my_val
    assert  _my_awesome_format(format_func, val) == expected

@pytest.mark.parametrize("format_function,expected", [
    (_format_TM, 'correctanser'),
    (_format_DA, 'correctanser'),
...
])
def test_my_awesome_formating_multivalue(format_func, expected):
    val = [ my_val1, my_val2, my_val3 ]
    assert  _my_awesome_format(format_func, val) == expected

```
Or you can even combine both tests using a triplet: function, val, answer